app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala

```diff
@@ -38,35 +39,45 @@ class SuppressCaseController @Inject()(verify: RequestActions,
                                        val messagesApi: MessagesApi,
                                        implicit val appConfig: AppConfig) extends RenderCaseAction {
 
-  private val form: Form[Boolean] = MandatoryBooleanForm.form("suppress_case")
+  private val form: Form[String] = AddNoteForm.form
 
   override protected val config: AppConfig = appConfig
   override protected val caseService: CasesService = casesService
 
   override protected def redirect: String => Call = routes.CaseController.applicationDetails
   override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = c.status == NEW
 
-  private def showCase(reference: String, f: Form[Boolean])
-                      (implicit request: AuthenticatedCaseRequest[AnyContent]): Future[Result] = {
-    getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, f)))
-  }
-
   def getSuppressCase(reference: String): Action[AnyContent] = (verify.authenticated andThen verify.casePermissions(reference) andThen
     verify.mustHave(Permission.SUPPRESS_CASE)).async { implicit request =>
-    showCase(reference, form)
+    getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, form)))
   }
 
-  def postSuppressCase(reference: String): Action[AnyContent] = (verify.authenticated andThen verify.casePermissions(reference) andThen
-    verify.mustHave(Permission.SUPPRESS_CASE)).async { implicit request =>
+  def postSuppressCase(reference: String): Action[MultipartFormData[Files.TemporaryFile]] =
+    (verify.authenticated andThen verify.casePermissions(reference) andThen verify.mustHave(Permission.SUPPRESS_CASE))
+      .async(parse.multipartFormData) { implicit request: AuthenticatedCaseRequest[MultipartFormData[Files.TemporaryFile]] =>
 
-    form.bindFromRequest().fold(
-      errors => showCase(reference, errors),
-      {
-        case true => validateAndRedirect(casesService.suppressCase(_, request.operator).map(c => routes.SuppressCaseController.confirmSuppressCase(reference)))
-        case _ => validateAndRedirect(c => successful(routes.SuppressCaseController.showContactInformation(reference)))
-      }
-    )
+    request.body.file("email").filter(_.filename.nonEmpty).filter(_.contentType.isDefined) match {
+      case Some(file) =>
+        form.bindFromRequest().fold(
+          errors =>
+            getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, errors))),
```

Comment:
I'm facing the same issue on my change - but when you go down this route (you added a valid file but no note), when the page reloads, the file you choose is no longer selected, but no error shows, so then you can't click on confirm successfully until picking a file again.

Reply:
I think its got to be like this for security reasons :( We can pre-populate the note but not the file

Reply:
Fair enough, it's a minor annoyance at most I guess anyway.